pkg/trace/writer/stats.go

```diff
@@ -95,6 +109,29 @@ func (w *StatsWriter) Run() {
 	}
 }
 
+// FlushSync is a no-op for TraceWriter
+func (w *StatsWriter) FlushSync() error {
+	if !w.syncMode {
+		return errors.New("SyncFlush called on StatsWriter, which is a no-op")
+	}
+
+	defer w.report()
+	notify := make(chan struct{}, 1)
+	w.flushChan <- notify
+	<-notify
+
+	var wg sync.WaitGroup
+	for _, s := range w.senders {
+		wg.Add(1)
+		go func(s *sender) {
+			defer wg.Done()
+			s.waitForInflight()
```

Comment:
Please make this made exported (since it's obviously not private anymore) and document it. `WaitForInflight` is a good name.